### PR TITLE
Scenario tests for dropped databases on Managed Instance

### PR DESCRIPTION
**Pull request by v-djnisi**
## Description

This are scenario tests for dropped databases. I added test testing gets for dropped databases, as well as update and get for short term retention on dropped databases.

Rest-api-specs: [https://github.com/Azure/azure-rest-api-specs/pull/4953](url)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs
Comment (dsgouda) [inline src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs]: Please pull latest changes from psSdkJson6, run `git clean -xdff` followed by `msbuild build.proj` and regenerate the code

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
@v-djnisi the rest spec PR is inaccessible, please fix the link

**Comment by v-djnisi** [comment]
[https://github.com/Azure/azure-rest-api-specs/pull/4953](url)

File: src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs
Comment (v-djnisi) [reply src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs]: I tried this, as well as repeating the whole process. But it didn't generate this part of the code.

**Comment by v-djnisi** [comment]
I ran the same command but from ps file, and it somehow worked. Everything should be okay now.

**Comment by dsgouda** [comment]
Thanks @v-djnisi will approve and merge once the REST spec PR is merged

**Comment by v-djnisi** [comment]
Rest-api-specs was merged, regenerated synced code and squshed.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**
